x509-cert/src/serial_number.rs

```diff
@@ -91,6 +92,27 @@ impl Display for SerialNumber {
     }
 }
 
+macro_rules! impl_from {
+    ($source:ty) => {
+        impl From<$source> for SerialNumber {
+            fn from(inner: $source) -> SerialNumber {
+                let serial_number = &inner.to_be_bytes()[..];
+                let serial_number = asn1::Uint::new(serial_number).unwrap();
```

Comment:
@woodruffw The `asn1::Int` API introduced in #823 is bit deceiving I think.
I can't just feed it the big endian representation of an int directly and I have to filter that out first.

I think I want to try something in the idea of:
```
diff --git a/der/src/asn1/integer/bigint.rs b/der/src/asn1/integer/bigint.rs
index c5d7ff8..2c43609 100644
--- a/der/src/asn1/integer/bigint.rs
+++ b/der/src/asn1/integer/bigint.rs
@@ -22,7 +22,7 @@ pub struct IntRef<'a> {
 impl<'a> IntRef<'a> {
     /// Create a new [`IntRef`] from a byte slice.
     pub fn new(bytes: &'a [u8]) -> Result<Self> {
-        let inner = BytesRef::new(int::strip_leading_ones(bytes))
+        let inner = BytesRef::new(uint::strip_leading_zeroes(int::strip_leading_ones(bytes)))
             .map_err(|_| ErrorKind::Length { tag: Self::TAG })?;

         Ok(Self { inner })
```

Reply:
Yep, agreed that it's deceptive. 

That said, I think stripping both will cause other problems: without some additional state for the signedness, it'll be impossible to re-encode the integer accurately. 

We've been discussing that in #826 and I think @tarcieri's last suggestion makes sense; I'll have time to implement it tomorrow most likely 🙂